keps/77-dynamically-sized-jobs/README.md

```diff
@@ -7,168 +7,290 @@
 ```golang
+// Workload slicing can be enabled for a specific integration job instance,
+// provided that the integration supports workload slicing.
 const (
-  ...
-  ...
-	// WorkloadResizeRequested means that the Workload is in the process of scaling up or down
-	WorkloadResizeRequested = "ResizeRequested"
+  // EnabledAnnotationKey refers to the annotation key present on Job's that support
+  // workload slicing.
+  // This annotation is alpha-level.
+  EnabledAnnotationKey = "kueue.x-k8s.io/elastic-job"
+  // EnabledAnnotationValue refers to the annotation value. To enable
+  // workload slicing for a given job, we match both annotation key and value.
+  EnabledAnnotationValue = "true"
 )
-
 ```
 
-### Creating Workload Slices
+### Processing of Workload Slices
+
+#### Creation
+The creation of WorkloadSlices in Kueue is initiated when a Kueue Job with slicing enabled, via both annotation and feature gate, is admitted for execution. 
+Upon admission, the `jobframework.JobReconciler` evaluates the Job’s PodSets and resource requirements, then creates a new or additional (slice) Workload object 
+that represents a partition of the original workload. Each slice contains a `PodSet.Count` identical to the Job’s definition, along with scheduling metadata needed for handling preemption of the old workload (slice).
 
-The [GenericJob interface (7e778f5)](https://github.com/kubernetes-sigs/kueue/blob/main/pkg/controller/jobframework/interface.go#L30-L55) will be updated to handle resize operations of jobs.
+At most two active workload slices can exist for any single Job in Kueue: the current admitted slice and a newly created slice during a scaling operation. 
+This ensures a controlled transition between slices while maintaining consistency and avoiding resource contention and race conditions.
+
+To facilitate precise resource scheduling and effective workload preemption, every new job configured with slice enablement will be appropriately annotated using `EnabledAnnotationKey/Value` mentioned above.
+Furthermore, each individual workload slice will record the identity of any preempting slice it encounters.
 
 ```golang
-type GenericJob interface {
-  ...
-  ...
-  ResizeJob(wl *kueue.Workload) error
-}
+
+const (
+	// WorkloadPreemptibleSliceNameKey is the annotation key used to capture an "old" workload slice
+	// that will be preempted by the "new" workload slice, i.e., this annotates a "new" workload slice.
+	// This annotation is alpha-level.
+	WorkloadPreemptibleSliceNameKey = "kueue.x-k8s.io/workload-preemptible-slice"
+)
 ```
-Jobs implementing the ResizeJob method will create a Workload Slice for every new replica of a job. 
 
-On scale down to M we will change the original Workload's resources and then on scale up to N we will create N-M WorkloadSlice objects that go through admission and scheduling gate checks, i.e. the Workload can "lose" the original (first time admission) quota when scaling down before scaling up.
+##### Naming
```

Comment:
When there are old and new Workload (slice) in the cluster, how to identify the latest one from batch/v1 Job object?

Reply:
I see. SGTM.
Could you mention that we always respect the latest created WorkloadSlice?